packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -1438,6 +1451,77 @@ export class SnapController extends BaseController<
     return encodeAuxiliaryFile(value, encoding);
   }
 
+  /**
+   * Processes dynamic permission request.
+   *
+   * @param snapId - The id of the Snap requesting permissions.
+   * @param permissions - Structure with requested permissions.
+   * @returns Result of permission grant request.
+   */
+  async requestPermissions(
```

Comment:
I'm wondering if more of this logic should be in the RPC method instead of the SnapController 🤔

Reply:
I was hoping that the RPC methods could be almost aliases for existing RPC methods, but just with some extra validation

Reply:
All logic from controller is now in RPC method implementation. Hook is used to communicate with PermissionController and provide approval flow.

Reply:
Nice!